dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java

```diff
@@ -221,14 +224,21 @@ private void addCollectionEntriesToCollection(Collection collection) {
         List<CollectionEntry> collectionEntries = new ArrayList<>();
         collectionEntries.addAll(collectionWorkflows);
         collectionEntries.addAll(collectionServices);
-        collectionEntries.addAll(collectionTools);
+        collectionEntries.addAll(collectionTools);        
         collectionEntries.addAll(collectionWorkflowsWithVersions);
         collectionEntries.addAll(collectionServicesWithVersions);
         collectionEntries.addAll(collectionToolsWithVersions);
         collectionEntries.forEach(entry -> {
             List<Label> labels = workflowDAO.getLabelByEntryId(entry.getId());
             List<String> labelStrings = labels.stream().map(Label::getValue).collect(Collectors.toList());
             entry.setLabels(labelStrings);
+            List<String> descriptorTypes;
+            if (entry.getEntryType().equals("tool")) {
+                descriptorTypes = toolDAO.getToolsDescriptorTypes(entry.getId());
+            } else {
```

Comment:
May be over-engineering, but I suggest explicitly checking what's allowed instead assuming it's OK if it's not a tool. If we add a new entry type some day, e.g., notebooks, line 239 could fail. And I think it would fail on a service, although we don't allow services to be added to collections.